cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	k8s "k8s.io/client-go/kubernetes"
+
+	"sigs.k8s.io/kueue/client-go/clientset/versioned/scheme"
+	"sigs.k8s.io/kueue/cmd/kueuectl/app/util"
+)
+
+const (
+	podLong = `Lists all pods that matches the given criteria: Should be part of the specified Job kind,
+	belonging to the specified namespace, matching
+	the label selector or the field selector.`
+	podExample = `  # List Pods
+  kueuectl list pods --for job/job-name`
+)
+
+type objectRef struct {
```

Comment:
Maybe it's a good idea to move it to `helper.go`. IIUC the same struct using on #2238

Reply:
Yup, inspired from that PR, I ll move those to `helpers.go`

Reply:
Moved in [b3eab73](https://github.com/kubernetes-sigs/kueue/pull/2280/commits/b3eab73f436a055029766f3cf84ad330b940825d)